File: src/main/java/org/jukeboxmc/block/BlockCaveVinesBodyWithBerries.java
package org.jukeboxmc.block;

import org.jukeboxmc.item.ItemCaveVinesBodyWithBerries;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class BlockCaveVinesBodyWithBerries extends Block{

    public BlockCaveVinesBodyWithBerries() {
        super( "minecraft:cave_vines_body_with_berries" );
    }

    @Override
    public ItemCaveVinesBodyWithBerries toItem() {
        return new ItemCaveVinesBodyWithBerries();
    }

    @Override
    public BlockType getBlockType() {
        return BlockType.CAVE_VINES_BODY_WITH_BERRIES;
    }

    @Override
    public boolean isTransparent() {
        return true;
    }

    @Override
    public boolean isSolid() {
        return false;
    }

    public void setGrowingPlantAge( int value ) {
        this.setState( "growing_plant_age", value );
    }

    public int getGrowingPlantAge() {
        return this.stateExists( "growing_plant_age" ) ? this.getIntState( "growing_plant_age" ) : 0;
    }
}

File: JukeboxMC-Server/src/main/kotlin/org/jukeboxmc/server/block/behavior/BlockCaveVinesBodyWithBerries.kt
package org.jukeboxmc.server.block.behavior

import org.cloudburstmc.nbt.NbtMap
import org.jukeboxmc.api.Identifier
import org.jukeboxmc.api.block.CaveVinesBodyWithBerries
import org.jukeboxmc.server.block.JukeboxBlock

class BlockCaveVinesBodyWithBerries(identifier: Identifier, blockStates: NbtMap?) :
    JukeboxBlock(identifier, blockStates),
    CaveVinesBodyWithBerries {

    override fun getGrowingPlantAge(): Int {
        return this.getIntState("growing_plant_age")
    }

    override fun setGrowingPlantAge(value: Int): BlockCaveVinesBodyWithBerries {
        return this.setState("growing_plant_age", value)
    }
}